neoload/neoload_cli_lib/schema_validation.py

```diff
@@ -53,4 +48,91 @@ def validate_yaml(yaml_file_path, schema_url, ssl_cert=''):
         for error in sorted(v.iter_errors(yaml_as_object), key=str):
             path = "\\".join(list(map(lambda x: str(x), error.path)))
             msgs += "\n" + error.message + "\n\tat: " + path + "\n\tgot: \n" + yaml.dump(error.instance) + "\n"
+        msgs = ("in file %s" % yaml_file_path) + msgs
         raise ValueError(YAML_NOT_CONFIRM_MESSAGE + '\n' + msgs)
+
+
+def validate_yaml_dir(path, schema_spec, ssl_cert='',continue_on_error=True):
+    ignore_file = os.path.join(path, '.nlignore')
+    nl_ignore_matcher = parse_gitignore(ignore_file) if os.path.exists(ignore_file) else None
+    first_time_check = True
+    extensions = ['yml','yaml','json']
+    any_errs = False
+    for root, dirs, files in os.walk(path):
+        for file in files:
+            file_path = os.path.join(root, file)
+            (any_errs,first_time_check) = validate_yaml_dir_file(file_path,schema_spec,extensions,nl_ignore_matcher,any_errs,first_time_check,continue_on_error)
+
+    if any_errs:
+        raise ValueError('One or more errors in files underneath this directory.')
+
+def validate_yaml_dir_file(file_path,schema_spec,extensions,nl_ignore_matcher,any_errs,first_time_check,continue_on_error):
+
+    if any(filter(lambda ext,file_path=file_path: file_path.endswith("."+ext),extensions)) and \
+       not is_not_to_be_included(file_path, nl_ignore_matcher):
+        logging.debug("file_path: {}".format(file_path))
+        first_time_check = False
```

Comment:
first_time_check = False.  Always, so if I don't have the scema file, then I can't run the validate command, it throws an error...

Reply:
Good catch. I've moved it after the except so that if no exceptions, it flags for the next time around.

Reply:
Good catch. I've moved it after the except so that if no exceptions, it flags for the next time around.